### PR TITLE
Improve explanations in GNN notebook example

### PR DESCRIPTION
**Pull request by adamnsch**
Thank you for your contribution to the Graph Data Science Client project.

Before submitting this PR, please read [Contributing to the Neo4j Ecosystem](https://github.com/neo4j/graph-data-science-client/blob/main/CONTRIBUTING.md). 

Make sure:
- [x] You signed the [Neo4j CLA](https://neo4j.com/developer/cla/#sign-cla) (Contributor License Agreement) so that we are allowed to ship your code in our library
- [x] Your contribution is covered by tests

### CODE REVIEW

File: examples/import-sample-export-gnn.ipynb [hunk 1]
```diff
@@ -106,7 +106,8 @@
    "source": [
     "## Sampling CORA\n",
     "\n",
-    "Next we use the built in CORA loader to get the data into GDS. We will then sample it to get a smaller graph to train on."
+    "Next we use the built in CORA loader to get the data into GDS. We will then sample it to get a smaller graph to train on.\n",
```
Comment (breakanalysis) [inline examples/import-sample-export-gnn.ipynb:109]: built-in ?
not sure

File: examples/import-sample-export-gnn.ipynb [hunk 2]
```diff
@@ -154,11 +182,63 @@
...
+    "# Let's see what we got:\n",
+    "display(sample_topology_df)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "a4ad967b",
+   "metadata": {},
+   "source": [
+    "We get the right amount of rows, one for each expected relationship. However, the node IDs are rather large whereas PyG expects consecutive IDs starting from zero. So we will have to handle that shortly. Next we should massage our data structure containing the relationships so that it's injectable into PyG."
```
Comment (breakanalysis) [inline examples/import-sample-export-gnn.ipynb:205]:
```suggestion
    "We get the right amount of rows, one for each expected relationship. However, the node IDs are rather large whereas PyG expects consecutive IDs starting from zero. We will start to massage our data structure containing the relationships until PyG can use it."
```

**Review by breakanalysis — APPROVED** [review]
LGTM, but left some suggestions anyway :)

File: examples/import-sample-export-gnn.ipynb (hunk 1 above)
Comment (adamnsch) [reply examples/import-sample-export-gnn.ipynb:109]: done

File: examples/import-sample-export-gnn.ipynb (hunk 2 above)
Comment (adamnsch) [reply examples/import-sample-export-gnn.ipynb:205]: done

File: examples/import-sample-export-gnn.ipynb [hunk 3]
```diff
@@ -275,22 +380,63 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "ff062e04",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# In standard PyTorch fashion we instantiate our model, and transfer its memory to the chosen device\n",
```
[the remark this answers is not in the document]
Comment (adamnsch) [reply examples/import-sample-export-gnn.ipynb]: done

**Pull request merged**